tests/testthat/test-dataset.R

```diff
@@ -11,6 +11,8 @@ with(fake.HTTP, {
         expect_identical(name(test.ds), "test ds")
         expect_identical(description(test.ds), "")
         expect_identical(id(test.ds), "511a7c49778030653aab5963")
+        expect_identical(start_date(test.ds), "2016-01-01") 
+        expect_identical(end_date(test.ds), "2016-01-01")
```

Comment:
Great! A couple more things I'd like to see before accepting this: 

* expectation for getting start_date and end_date for a dataset that doesn't have one set. What should it be? (I'd say `NULL`.) Use one of the other two dataset mocks in the dataset.json file for that. You may need to add the fields to the tuples in the JSON and make sure that you're putting in the values that the API gives for datasets that don't have start_date/end_date defined yet (I believe it is `null`, but confirm with your own datasets).
* Add test coverage for the setters. Within the fake.HTTP context you're in there, whenever you PATCH/PUT/POST, you get an error with the verb, URL, and payload, so you can assert that you're making the right request with the right payload. See https://github.com/Crunch-io/rcrunch/blob/c5acab7964849cf8194d5079d0ae5b3ee098310a/tests/testthat/test-versioning.R#L27-L30 for an example of this.